### PR TITLE
Add new linux installation

### PR DESCRIPTION
**Pull request by ann0see**
I haven't tested the `sudo apt-get install -f` on a clean system yet!

### CODE REVIEW

File: wiki/en/en-Installation-for-Linux.md
Comment (gilgongo) [inline wiki/en/en-Installation-for-Linux.md]: I thought we were in the Ubuntu repos?

**Comment by gilgongo** [comment]
This prosebot sure is spammy...

File: wiki/en/en-Installation-for-Linux.md
Comment (ann0see) [reply wiki/en/en-Installation-for-Linux.md]: No. We're in the Debian repos. Maybe this also adds us to the Ubuntu repo automatically?

Comment (gilgongo) [reply wiki/en/en-Installation-for-Linux.md]: Hm. I get "Unable to locate package jamulus" when I do apt install jamulus on Debian 10 anyway...

Comment (gilgongo) [reply wiki/en/en-Installation-for-Linux.md]: Oh wait it's only for Bullseye. OK. Very few people will be using that :-)

**Review by gilgongo — APPROVED** [review]

File: wiki/en/en-Installation-for-Linux.md
Comment (trebmuh) [reply wiki/en/en-Installation-for-Linux.md]: Bullseye will (probably) be the new Debian "stable" in a few months now.
Maybe it worth to put a warning on this while Bullseye isn't yet the "stable" version of Debian?

Comment (gilgongo) [reply wiki/en/en-Installation-for-Linux.md]: Bullseye doesn't have a [release date](https://wiki.debian.org/DebianReleases) listed yet, and I don't know when the Ubuntu repos will incorporate Bullseye software. So for most Linux users apt install isn't going to be a thing for quite while maybe. 

Also BTW I note [somebody saying](https://sourceforge.net/p/llcon/discussion/software/thread/aa3960d2e7/#08d4) that the dependencies for Fedora were wrong in the docs

Comment (trebmuh) [reply wiki/en/en-Installation-for-Linux.md]: > Bullseye doesn't have a [release date](https://wiki.debian.org/DebianReleases) listed yet, and I don't know when the Ubuntu repos will incorporate Bullseye software.

I know that. The purpose of my previous message was to give an order of magnitude.

Debian policy on this matter is "it's going to be released when it'll be released", so waiting for a release date will not help because it'll not be announced before D-day. That said, [the release process has already been started](https://release.debian.org/bullseye/freeze_policy.html) and, at this time (and without any warranty, this is a moving target), it's fair to think that we __should__ have a Debian Bullseye release around spring time / beginning of summer. Of course, something can always go wrong and the release would then be postponed.

About Ubuntu, that will depend what they will import from testing/sid, and when.

All of that to try to clarify that the purpose of my previous mail was: if we are to write that Debian/Ubuntu users can `apt install jamulus`, it'd be wise to add a note on it that it'll be working for people using [Debian Bulleyes](https://packages.debian.org/search?keywords=jamulus) or [Ubuntu Hirsute](https://packages.ubuntu.com/search?keywords=jamulus), not others for the moment.

**Review by gilgongo — COMMENTED** [review]
Sandwiching the compile section between the binary install method and the sound setup make this doc rather hard to read I think. I'm not too bothered though as the number of Jamulus Linux client users is tiny, but at some point we should move the compilation instructions to an INSTALL.md file (and pull it out of the server install docs as well) as is the general convention once people can just apt-get. 

EDIT: There's quite a lot that needs sorting out on this page yet though still.

File: wiki/en/en-Installation-for-Linux.md
Comment (pljones) [reply wiki/en/en-Installation-for-Linux.md]: For the time being, how easy would it be to automatically maintain an Ubuntu PPA?

Comment (gegeweb) [reply wiki/en/en-Installation-for-Linux.md]: FYI, I tried to compile/build the Jamulus deb from the Bullseye source package (this is usually trivial) on a Debian 10 buster.
There is an error:

```
RCC: Error in 'src/resources.qrc': Cannot find file 'res/translation/translation_de_DE.qm'
make[1]: *** [Makefile:1629: src/res/qrc_resources.cpp] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/gerald/sources/debian/jamulus-3.6.2+dfsg1'
dh_auto_build: error: make -j4 returned exit code 2
make: *** [debian/rules:45: binary] Error 25
dpkg-buildpackage: erreur: debian/rules binary subprocess returned exit status 2
debuild: fatal error at line 1182:
dpkg-buildpackage -us -uc -ui -b failed
```
The files on `src/res/translation` have a `*.ts` extension, not `*.qm`. The `*.qm`files aren't build.
 
Compiling and installing Jamulus on a Linux Debian based (Raspberry Pi OS Buster, 32 and 64 bits) following the instructions on install linux page works fine (for the client, and for the server).
I've installed Jamulus 3.6.2 (and upgraded from 3.6.0 and 3.6.1) on a Raspberry Pi OS 32 bits (server), on a Raspberry Pi OS 64 bits (client), both are a Debian 10 buster based system, and a Ubuntu 20.04 following the compilation instructions on the linux page.

My opinion is to say:
- if an official distribution package exists, install the package of the distribution according to the usual procedures on it.
- if not, compile and install Jamulus by following step by step the instructions on the Linux installation page.

Installing testing package on a stable Ubuntu or Debian is not recommended!

File: wiki/en/en-Installation-for-Linux.md
Comment (gegeweb) [inline wiki/en/en-Installation-for-Linux.md]: It is better to do:
`sudo apt install /path/to/jamulus_latest_ubuntu_amd64.deb`
This will automatically install the dependencies without need to do the `apt-get install -f`.

File: wiki/en/en-Installation-for-Linux.md
Comment (gegeweb) [inline wiki/en/en-Installation-for-Linux.md]: There is no *.deb package in the assets of latest release on the Github repo. Only the source code archives.

File: wiki/en/en-Installation-for-Linux.md
Comment (ann0see) [reply wiki/en/en-Installation-for-Linux.md]: Thanks. Can you apply the proposal here? You should have push access.

File: wiki/en/en-Installation-for-Linux.md
Comment (ann0see) [reply wiki/en/en-Installation-for-Linux.md]: Yes. But they will be there. See the beta release: https://github.com/jamulussoftware/jamulus/releases/tag/r_3_7_0beta1

Comment (trebmuh) [reply wiki/en/en-Installation-for-Linux.md]: > this is usually trivial

I disagree. Debian 10 (buster) and Debian 11 (Bulleyes) are 2 different distributions with 2 different toolchain and 2 different packages libraries versions. It's not that unusual that a simple package rebuild fails.


> My opinion is to say:
>  if an official distribution package exists, install the package of the distribution according to the usual procedures on it.
>  if not, compile and install Jamulus by following step by step the instructions on the Linux installation page.

Agreed.


> Installing testing package on a stable Ubuntu or Debian is not recommended!

It's a general Debian/Ubuntu advice. It looks to be outside of the Jamulus's project jurisdiction.

Comment (gegeweb) [reply wiki/en/en-Installation-for-Linux.md]: > > this is usually trivial
> 
> I disagree. Debian 10 (buster) and Debian 11 (Bulleyes) are 2 different distributions with 2 different toolchain and 2 different packages libraries versions. It's not that unusual that a simple package rebuild fails.

I have recompiled a lot of packages not available yet in Buster but available in testing from source.
It is really usualy trivial (it should be the case for jamulus) if all the deps build are available on Buster. And that is the case for Jamulus, everything needed to compile/build Jamulus on this system is available (as listed on the actual install linux page for the 3.6.2).

You have just to add the debian testing source (and only the source) to sources.list

```
# Uncomment deb-src lines below then 'apt-get update' to enable 'apt-get source'
deb-src http://deb.debian.org/debian buster main contrib non-free
deb-src http://deb.debian.org/debian-security/ buster/updates main contrib non-free
deb-src http://deb.debian.org/debian buster-updates main contrib non-free
# sources from testing/unstable 
deb-src http://httpredir.debian.org/debian unstable main
```

And then (for jamulus):

```
apt-get source jamulus
cd jamulus-3.6.2+dfsg1/
sudo apt-get build-dep jamulus # this will download and install the build deps from Buster packages
debuild -b -uc -us
cd ../
sudo apt install ./jamulus_3.6.2+dfsg1.arm64.deb # for example
```
I prefer using apt install instead of dpkg -i because apt install all the deps in one step.

What is very strange is that on this machine compiling from source works, not building the deb.
Something seems to be broken or missing in the deb source for the translations files.

>>  Installing testing package on a stable Ubuntu or Debian is not recommended!
> It's a general Debian/Ubuntu advice. It looks to be outside of the Jamulus's project jurisdiction.

Yes but adding testing or unstable repo in sources.list to install Jamulus with apt on a stable Debian or Ubuntu is not a good idea.

It is better (and it works fine following the doc) to install from sources if no official distribution (or team) package available. (IMHO)

Comment (trebmuh) [reply wiki/en/en-Installation-for-Linux.md]: > I have recompiled a lot of packages not available yet in Buster but available in testing from source.

It **can/might** work sometime but it's not to be recommanded to the end users in any way.

(BTW, I know how a backport works)

Comment (ann0see) [reply wiki/en/en-Installation-for-Linux.md]: I think a backport is already in preparation: https://github.com/jamulussoftware/jamulus/issues/896#issuecomment-767631624
let's just wait for this.

Comment (trebmuh) [reply wiki/en/en-Installation-for-Linux.md]: Much better to tell the end users to use a package from buster-backports rather than backporting the package himself.

File: wiki/en/en-Installation-for-Linux.md
Comment (ann0see) [reply wiki/en/en-Installation-for-Linux.md]: Ok. Just applied this proposal.

File: wiki/en/en-Installation-for-Linux.md
Comment (pljones) [inline wiki/en/en-Installation-for-Linux.md]: "is will be" - pick either "is" or "will be", not both :)

Comment (ann0see) [reply wiki/en/en-Installation-for-Linux.md]: Fixed

**Review by pljones — APPROVED** [review]

**Comment by ann0see** [comment]
@gilgongo if you think it's ready, you can merge it.

File: wiki/en/en-Installation-for-Linux.md
Comment (gilgongo) [inline wiki/en/en-Installation-for-Linux.md]: Oops, "the above"

Comment (ann0see) [reply wiki/en/en-Installation-for-Linux.md]: Ok. Done.

**Review by gilgongo — APPROVED** [review]
Heh - it's even more complicated than it was before we had binaries. But no matter at the least the information's there.

**Comment by ann0see** [comment]
> it's even more complicated than it was before

This doesn't sound good. Any ideas on how to make it easier?

**Comment by gilgongo** [comment]
"Interleaving" binary install instructions with source compilation, then having "if / then" references to each after that is not ideal :-) But it's not big deal really. Once we're in the current versions of Ubuntu and Debian (and maybe other?) distro's repos, we can follow the convention of kicking compilation instructions into a INSTALL.md for the tiny number of people who would want to do that. 

BTW it's a pity we don't have a jamlus-server package, as we'd be able to do the same for server operators. Can't have everything I guess.

**Comment by ann0see** [comment]
Yes. The headless .deb file might partly solve it.

File: wiki/en/en-Installation-for-Linux.md
Comment (pljones) [inline wiki/en/en-Installation-for-Linux.md:117]: > better performance

Should really say "lower latency" -- better performance is very vague.  What's "better" about more XRUNs, if that's what it causes someone?  Indeed, it's more that it requires a higher specification system - i.e. demands better performance, rather than provides it.

Anyway... for another time :).

**Pull request merged**
